packages/lib/src/validations/Constructors.ts

```diff
@@ -1,4 +1,17 @@
-// TS-TODO: Web3 typings?
-export function hasConstructor(contract: any): boolean {
-  return !!contract.schema.abi.find(fn => fn.type === 'constructor');
+import Contract from '../artifacts/Contract';
+import { BuildArtifacts } from '..';
+import ContractAST, { FunctionDefinitionFilter } from '../utils/ContractAST';
+
+export function hasConstructor(contract: Contract, buildArtifacts: BuildArtifacts): boolean {
+  return new ContractAST(contract, buildArtifacts, FunctionDefinitionFilter)
+    .getLinearizedBaseContracts()
+    .filter(hasNonEmptyConstructorInAST)
+    .length > 0;
+}
+
+function hasNonEmptyConstructorInAST(contractNode: any): boolean {
```

Comment:
We should also check that there are no modifiers!

Reply:
Good catch! Added, please re-review!